src/Riok.Mapperly/Descriptors/MappingBodyBuilders/ObjectMemberMappingBodyBuilder.cs

```diff
@@ -135,6 +135,27 @@ PropertyMappingConfiguration config
             return false;
         }
 
+        // cannot assign to intermediate value type
+        if (targetMemberPath.Path.Count > 1)
+        {
+            var member = targetMemberPath.Path.Reverse().Skip(1).First();
+            if (member.Type is { IsValueType: true, IsRefLikeType: false })
+            {
+                ctx.BuilderContext.ReportDiagnostic(
+                    DiagnosticDescriptors.CannotMapToTemporarySourceMember,
+                    ctx.Mapping.SourceType,
+                    sourceMemberPath.FullName,
+                    sourceMemberPath.Member.Type,
+                    ctx.Mapping.TargetType,
+                    targetMemberPath.FullName,
+                    targetMemberPath.Member.Type,
+                    member.Name,
+                    member.Type
+                );
+                return true;
```

Comment:
As discussed in https://github.com/riok/mapperly/issues/516#issuecomment-1655677725 it would make sense to return false here and not generate this mapping at all, wouldn't it?